sdk/kiosk/src/constants.ts

```diff
@@ -1,10 +1,90 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-/** The Transer Policy Rules package address on testnet */
-export const TESTNET_RULES_PACKAGE_ADDRESS =
-	'bd8fc1947cf119350184107a3087e2dc27efefa0dd82e25a1f699069fe81a585';
+import {
+	resolveFloorPriceRule,
+	resolveKioskLockRule,
+	resolvePersonalKioskRule,
+	resolveRoyaltyRule,
+} from './tx/rules//resolve';
+import { Network, type ObjectArgument, type RuleResolvingParams } from './types';
 
-/** The transfer policy rules package address on mainnet */
-export const MAINNET_RULES_PACKAGE_ADDRESS =
-	'0x434b5bd8f6a7b05fede0ff46c6e511d71ea326ed38056e3bcd681d2d7c2a7879';
+/**
+ * The Transfer Policy rule.
+ */
+export type TransferPolicyRule = {
+	rule: string;
+	network: Network;
+	packageId: string;
+	resolveRuleFunction: (rule: RuleResolvingParams) => ObjectArgument | void;
+	hasLockingRule?: boolean;
+};
+
+export const royaltyRuleAddress: Record<Network, string> = {
+	[Network.TESTNET]: 'bd8fc1947cf119350184107a3087e2dc27efefa0dd82e25a1f699069fe81a585',
+	[Network.MAINNET]: '0x434b5bd8f6a7b05fede0ff46c6e511d71ea326ed38056e3bcd681d2d7c2a7879',
+};
+
+export const kioskLockRuleAddress: Record<Network, string> = {
+	[Network.TESTNET]: 'bd8fc1947cf119350184107a3087e2dc27efefa0dd82e25a1f699069fe81a585',
```

Comment:
I feel like we need devnet versions for all of these

Reply:
@Jordan-Mysten It'll be difficult to maintain due to weekly wipes (we'd need to keep patching the address or come up with a remote solution to fetch it.

Reply:
Understood but I think it’s a requirement given we know folks will be developing against devnet.

Reply:
@Jordan-Mysten there's no way for us to hardcode a package on a network that's being reset weekly. We don't have a stable publish flow for none of our packages nor apps. Like if you want to build with SuiFrens you're forced to use testnet and only testnet. 

If someone wants to build against devnet, they're free to publish the package and specify a custom config for the application.

Reply:
I added a suggestion above about this, I think accepting these values directly when constructing a kiosk client would allow this to work in devnet without having to hard code them in the SDK.   It's not ideal, and better would be having a way to load them from something on chain, but this might be a decent short term compromise.  We can still use these constants when the network in mainnet or testnet